CHANGELOG.md

```diff
@@ -8,6 +8,65 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
+- `WordPress.PHP.DiscourageGoto` sniff to the `WordPress-Core` ruleset.
+- `WordPress.PHP.RestrictedFunctions` sniff to the `WordPress-Core` ruleset which initially forbids the use of `create_function()`.
+    This was previous only discouraged under certain circumstances.
+- `WordPress.WhiteSpace.ArbitraryParenthesesSpacing` sniff to the `WordPress-Core` ruleset which checks the spacing on the inside of arbitrary parentheses.
+- `WordPress.WhiteSpace.PrecisionAlignment` sniff to the `WordPress-Core` ruleset which will throw a warning when precision alignment is detected in PHP, JS and CSS files.
+- `WordPress.WhiteSpace.SemicolonSpacing` sniff to the `WordPress-Core` ruleset which will throw a (fixable) error when whitespace is found before a semi-colon, except for when the semi-colon denotes an empty `for()` condition.
+- `WordPress.CodeAnalysis.AssignmentInCondition` sniff to the `WordPress-Extra` ruleset.
+- `WordPress.WP.DiscouragedConstants` sniff to the `WordPress-Extra` and `WordPress-VIP` rulesets to detect usage of deprecated WordPress constants, such as `STYLESHEETPATH` and `HEADER_IMAGE`.
+- Ability to pass the `minimum_supported_version` to use for the `DeprecatedFunctions`, `DeprecatedClasses` and `DeprecatedParameters` sniff in one go. You can pass a `minimum_supported_wp_version` runtime variable for this [from the command line or pass it using a `config` directive in a custom ruleset](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#setting-minimum-supported-wp-version-for-all-sniffs-in-one-go-wpcs-0140).
+- `Generic.Formatting.MultipleStatementAlignment` - customized to have a `maxPadding` of `40` -, `Generic.Functions.FunctionCallArgumentSpacing` and `Squiz.WhiteSpace.ObjectOperatorSpacing` to the `WordPress-Core` ruleset.
+- `Squiz.Scope.MethodScope`, `Squiz.Scope.MemberVarScope`, `Squiz.WhiteSpace.ScopeKeywordSpacing`, `PSR2.Methods.MethodDeclaration`, `Generic.Files.OneClassPerFile`, `Generic.Files.OneInterfacePerFile`, `Generic.Files.OneTraitPerFile`, `PEAR.Files.IncludingFile`, `Squiz.WhiteSpace.LanguageConstructSpacing`, `PSR2.Namespaces.NamespaceDeclaration` to the `WordPress-Extra` ruleset.
+- The `is_class_constant()`, `is_class_property` and `valid_direct_scope()` utility methods to the `WordPress\Sniff` class.
+
+### Changed
+- When passing an array property via a custom ruleset to PHP Codesniffer, spaces around the key/value are taken as intentional and parsed as part of the array key/value. In practice, this leads to confusion and WPCS does not expect any values which could be preceded/followed by a space, so for the WordPress Coding Standard native array properties, like `customAutoEscapedFunction`, `text_domain`, `prefixes`, WPCS will now trim whitespace from the keys/values received before use.
```

Comment:
"PHP Codesniffer" => "PHP_CodeSniffer"

Reply:
Just wondering: AFAICS the package name is "PHP_Codesniffer", but the product name is "PHP Codesniffer", so why should the package name be used here instead of the product name ? or am I missing something ?

Reply:
The product name is also called `PHP_CodeSniffer` (underscore, capital S) - see the PHPCS [Readme](https://github.com/squizlabs/PHP_CodeSniffer), the PHPCS [wiki](https://github.com/squizlabs/PHP_CodeSniffer/wiki), the SquizLabs [blog posts](https://www.squizlabs.com/php-codesniffer/), `phpcs --version` etc. 

The package is `squizlabs/php_codesniffer`, but everywhere else official it is referred to as exactly `PHP_CodeSniffer`, and not `PHP Codesniffer`.

Reply:
Okidoki. Changed all pertinent references to PHPCS which were misspelled (in Readme/Changelog/Contributing).

Interestingly enough, in some code which is literally taken from PHPCS itself (test suite files for 2.x test support), the name is set as "PHP CodeSniffer"... 
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/615323e260492c6192bda208d390dab7158aa937/Test/AllTests.php#L53
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/615323e260492c6192bda208d390dab7158aa937/Test/Standards/AllSniffs.php#L58

As those are copied-in files, I haven't changed those.